addons/mail/static/src/new/composer/composer.js

```diff
@@ -421,7 +421,12 @@ export class Composer extends Component {
     onFocusin() {
         this.props.composer.isFocused = true;
         if (this.props.composer.thread) {
-            this.threadService.markAsRead(this.props.composer.thread);
+            const isChatterThread = this.props.composer.thread.type === "chatter";
+            if (isChatterThread) {
+                this.threadService.markAllMessagesAsRead(this.props.composer.thread);
+            } else {
+                this.threadService.markAsRead(this.props.composer.thread);
```

Comment:
Shouldn't both be used for the channel case?
To have needaction messages marked as read, at well as changing the "channel seen" status.

Reply:
Or I don't remember but didn't we make an optimization so that marking as read was coupled with fetching messages in python? Rather than requiring a separate rpc like this.

Reply:
There is indeed an optimization that was not applied because the `min` parameter was always passed to the route. In the case of failures, the message was known from inbox even if the origin thread was not fetched. I introduced the `isLoaded` concept (the one we use on master) to check whether or not the `min` parameter should be passed to the fetch message route and it works as expected.